src/WebJobs.Extensions.DurableTask/DefaultPlatformInformationProvider.cs

```diff
@@ -49,6 +44,60 @@ public bool InLinuxAppService()
             return inLinuxDedicated;
         }
 
+        public OperatingSystem GetOperatingSystem()
+        {
+            if (this.InLinuxConsumption() || this.InLinuxAppService())
+            {
+                return OperatingSystem.Linux;
+            }
+
+            return OperatingSystem.Windows;
+        }
+
+        public AppServicePlan GetAppSevicePlan()
+        {
+            if (this.InLinuxConsumption() | this.InWindowsConsumption())
+            {
+                return AppServicePlan.Consumption;
+            }
+
+            return AppServicePlan.AppService;
+        }
+
+        public ProgLanguage GetProgLanguage()
+        {
+            string workerRuntime = this.nameResolver.Resolve("FUNCTIONS_WORKER_RUNTIME");
+            if (workerRuntime == "python")
+            {
+                return ProgLanguage.Python;
+            }
+            else if (workerRuntime == "node")
+            {
+                return ProgLanguage.JavaScript;
+            }
+            else if (workerRuntime == "powerShell")
+            {
+                return ProgLanguage.PowerShell;
+            }
+            else if (workerRuntime == "dotnet")
+            {
+                return ProgLanguage.Csharp;
+            }
+
+            throw new Exception("Could not determine user-level programming language.");
+        }
+
+        public bool IsOutOfProc()
+        {
+            ProgLanguage progLanguage = this.GetProgLanguage();
+            if (progLanguage != ProgLanguage.Csharp)
```

Comment:
.NET Isolated is also out-of-proc.

Reply:
I think I'll remove this method in that case, as I can just do a more enum comparison instead